app/views/admin/shared_content/index.html.erb

```diff
@@ -0,0 +1,36 @@
+<% @page_title = t( 'admin.shared_content.shared_content' ) %>
+
+<%= form_with name: 'all_shared_content', url: admin_shared_content_path, method: :post do %>
+<table class="list">
+  <% @elements.each do |e| %>
+  <tr class="showfocus">
+    <td>
+      <%= text_field :shared_content, "element_#{e.id}_name", value: e.name %>
+      <%= select :shared_content, "element_#{e.id}_content_type", SharedElement.content_types, value: e.content_type %>
+      <br><%= text_field :shared_content, "element_#{e.id}_content", value: e.content %>
+    </td>
+    <td class="actions">
+      <%= link_to t( 'admin.delete' ), admin_shared_content_delete_path( e ),
+          method: :delete, data: { confirm: t( 'are_you_sure' ) } %>
+		</td>
+  </tr>
+  <% end %>
+</table>
+<p>
+  <%= submit_tag t( 'admin.update' ) %>
+</p>
+<% end  # form %>
+
+<%= form_with name: 'new_element', url: admin_shared_content_create_path, method: :post do %>
```

Comment:
Check if this form is friendly and remembers last entered values when the user submits invalid data. If it doesn't and it should, consider:

- providing a `model: @shared_element` option to the `form_with` call
- using the FormBuilder object (often named `f`) that `form_with` yields to its block `form_with ... do |f| ... end`. Use `f` to build the fields, e.g. `f.text_field ...`.
- modifying the relevant controller actions to support this.